### PR TITLE
Support naming and keying transit vehicle overlay

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
**Description:**
Previously there was no way to name/key the transit vehicle overlay. This adds a new way to do that without needing any new explicit configuration types!

### CODE REVIEW

File: lib/components/map/default-map.tsx [hunk 1]
```diff
@@ -341,7 +345,13 @@ class DefaultMap extends Component {
           {/* The default overlays */}
           <EndpointsOverlay />
           <RouteViewerOverlay />
-          <TransitVehicleOverlay ModeIcon={ModeIcon} />
+          <TransitVehicleOverlay
+            id={routeBasedTransitVehicleOverlayNameOverride || undefined}
```
Comment (daniel-heppner-ibigroup) [inline lib/components/map/default-map.tsx]: maybe move the `|| undefined` into the const so you don't have to repeat it on each of these!

**Review by daniel-heppner-ibigroup — APPROVED** [review]
LGTM. just one suggestion

**Review by amy-corson-ibigroup — APPROVED** [review]
Looks good!

**Comment by amy-corson-ibigroup** [comment]
Only thing is that percy is failing but I'm guessing you already noticed that!!

**Pull request merged**